### ISSUE TITLE
Show dependencies for library and executables separately

### ISSUE THREAD

**Issue by ChShersh** [haskell/hackage-server#789]
When I look at the package on Hackage and I want to compare this package with other packages, I evaluate number of dependencies of the library as a part of comparing process. Unfortunately, this is not an easy thing to do, because currently Hackage shows dependencies for both library and executable in the same section. Consider `life-sync` package:

* http://hackage.haskell.org/package/life-sync

Currently dependencies for this package look like this:

```haskell
Dependencies | ansi-terminal, 
base-noprelude (>=4.9 && <5), bytestring, containers, exceptions, fmt, life-sync, 
microlens-platform, optparse-applicative, path, 
path-io, process, relude (>=0.1.0), text (>=1.2), tomland (==0.3.*)
```

So the `Dependencies` section contains dependencies for both library and executable. If I press the `details` button I can see separation between library and executable. But I'm looking at Hackage for 3 years and noticed the `details` only now. It would be really great if the separation between package components was clearer. For example something like this:

```haskell
Dependencies
    library (62)
        ansi-terminal, base-noprelude (>=4.9 && <5), bytestring, containers, 
        exceptions, fmt, microlens-platform, path, path-io, process, 
        relude (>=0.1.0), text (>=1.2), tomland (==0.3.*)

    executable:life (64)
        base-noprelude, containers, life-sync, optparse-applicative, path
```

**Comment by phadej** [comment]
This will become even more important with multiple public libraries. I.e. show deps per component.

**Comment by hvr** [comment]
This should be co-designed with the rendering of the upcoming per-component `description`/`synopsis` fields

**Comment by amigalemming** [comment]
Currently, in this example the dependencies of the test-suite are not shown at all:
https://hackage.haskell.org/package/lapack-0.2.1/dependencies
:-(
Maybe i should open a new issue ...

**Comment by hvr** [comment]
@amigalemming yeah, that seems like a actionable-right-now task which doesn't require design work
